packages/components/src/text-control/index.tsx

```diff
@@ -16,6 +16,13 @@ import BaseControl from '../base-control';
 import type { WordPressComponentProps } from '../ui/context';
 import type { TextControlProps } from './types';
 
+function useUniqueId( idProp?: string ) {
+	const instanceId = useInstanceId( TextControl );
```

Comment:
FYI we can pass the id prefix and the `idProp` directly into the `useInstanceId()` function to avoid this custom code ✨

https://github.com/WordPress/gutenberg/blob/10d927c2654cadf850fd1930cd03090cbab1cd0a/packages/compose/src/hooks/use-instance-id/index.ts#L40-L48

Reply:
I noticed this, and strongly considered using it. But the `useUniqueId` pattern seemed to be in quite a few places already, which made me second guess myself.

I'm quite happy to apply `useInstanceId` "properly" here though, if that's actually the more appropriate direction.

Reply:
> I'm quite happy to apply `useInstanceId` "properly" here though, if that's actually the more appropriate direction.

Yes, I think it is. I have a feeling that the existing `useUniqueId` patterns were written before `useInstanceId` was expanded to take three arguments.